src/merkle.rs

```diff
@@ -0,0 +1,109 @@
+use sha3::{Digest, Sha3_256};
+
+type Hash = [u8; 32];
+
+pub struct MerkleTree {
+    tree: Vec<Vec<Hash>>,
+}
+
+fn concat_hash(left: &Hash, right: &Hash) -> Hash {
+    let mut hasher = Sha3_256::new();
+    hasher.update(left);
+    hasher.update(right);
+    hasher.finalize().into()
+}
+
+fn hash(value: &str) -> Hash {
+    Sha3_256::digest(value).into()
+}
+
+impl MerkleTree {
+    pub fn new(data: Vec<&str>) -> MerkleTree {
+        let mut merkle = MerkleTree { tree: Vec::new() };
+        if data.is_empty() {
+            return merkle;
+        }
+
+        // push hashes of the input
+        let leafs: Vec<Hash> = data.iter().map(|value| hash(value)).collect();
```

Comment:
```suggestion
        let leafs: Vec<Hash> = data.iter().map(hash).collect();
```
Here the clousure is not needed

Reply:
It doesn't compile :(

Reply:
https://www.youtube.com/watch?v=CGj85pVzRJs